assembl/static2/js/app/components/cookies/cookiesSelector.jsx

```diff
@@ -0,0 +1,96 @@
+// @flow
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Button } from 'react-bootstrap';
+import classnames from 'classnames';
+import CookieToggle from './cookieToggle';
+import type { CookieObject } from './cookieToggle';
+
+type CookiesSelectorProps = {};
+
+type CookiesSelectorState = {
+  activeKey: ?string,
+  show: boolean,
+  cookies: Array<CookieObject>
+}
+
+
+class CookiesSelector extends React.Component<CookiesSelectorProps, CookiesSelectorState> {
+  constructor(props: CookiesSelectorProps) {
+    super(props);
+    const cookiesList = document.cookie.split(' ');
+    const cookies = [];
+    cookiesList
+      .map(cookie => ({ ...this.getCookieObject(cookie), accepted: true }))
+      .forEach(cookieObject => cookies.push(cookieObject));
+    this.state = {
+      activeKey: 'essential',
+      show: true,
+      cookies: cookies
+    };
+  }
+
+  getCookieObject = (cookie: string) => {
+    if (cookie.includes('assembl_session')) {
+      return { category: 'essential', name: 'userSession' };
+    }
+    if (cookie.includes('LOCALE')) {
+      return { cateogry: 'essential', name: 'locale' };
+    }
+    if (cookie.includes('_pk_')) {
+      return { category: 'analytics', name: 'piwik' };
+    }
+    return { category: 'other', name: cookie };
+  };
+
+  handleToggle = (updatedCookie: CookieObject) => {
+    const { cookies } = this.state;
+    const filteredCookies = cookies.filter(c => c.name !== updatedCookie.name);
+    filteredCookies.push(updatedCookie);
```

Comment:
I prefer `const filteredCookies = cookies.map(c => c.name === updatedCookie.name ? updatedCookie : c)` to do this in one line and without mutability (but it's just a matter of preference)